diceParser.py

```diff
@@ -1,142 +1,144 @@
 #!/usr/bin/env python3
 
-#module functions
 import re
 import random
 
-def parse( diceString ):
 
-    #check if diceString is a string
+def parse(diceString):
+    # Check if diceString is a string
     if not isinstance(diceString, str):
         print("The provided input must be a string")
         return
 
-    #strips trailing and leading whitespace from input
+    # Strips trailing and leading whitespace from input
     diceString = diceString.strip().lower()
 
-    #check if valid string
+    # Check if valid string
     if not re.match('^[0-9]*[dD][1-9][0-9]*\s*([+-][0-9]*)?\s*([kKdD][lLhH][0-9]*)?$', diceString):
         print("The provided string is not a valid dice role")
         return
 
-    #get the ammount of dice
-    ammount = re.search('[0-9]+[dD]', diceString)
-    if ammount:
-        print("Number of dice: %s" % (ammount.group(0)[:-1]))
-        ammount = int(ammount.group(0)[:-1])
+    # Get the amount of dice
+    amount = re.search('[0-9]+[dD]', diceString)
+    if amount:
+        print("Number of dice: %s" % (amount.group(0)[:-1]))
+        amount = int(amount.group(0)[:-1])
     else:
-        ammount = 1;
+        amount = 1
 
-    #get the type of dice
+    # Don't roll more than 10,000 dice
+    if amount > 10000:
+        print("Too many dice, exiting")
+        return
+
+    # Get the type of dice
```

Comment:
Should replace with a Timeout in the roll function. This will be fine for now though